store/tikv/tests/lock_test.go

```diff
@@ -640,3 +643,65 @@ func (s *testLockSuite) TestBatchResolveTxnFallenBackFromAsyncCommit(c *C) {
 	_, err = t3.Get(context.Background(), []byte("fb2"))
 	c.Assert(tikverr.IsErrNotFound(err), IsTrue)
 }
+
+func (s *testLockSuite) TestDeadlockReportWaitChain(c *C) {
```

Comment:
Could the full wait chain be reported or showed by now? If so maybe we could add a more complex test case with a deadlock cycled graph.

Reply:
Yes the full wait chain can be reported now.

Reply:
It's kind of difficult to construct too complicated test case here since the `LockKeys` function locks a mutex and therefore two invocation on `LockKeys` can't run at the same time to make a transaction blocked by multiple blockers.  But there should be some way, I'm trying.